test/performance/range/views/view_translate_2D_1D_benchmark.cpp

```diff
@@ -0,0 +1,227 @@
+//  sequential_read
+// ============================================================================
+
+template <typename rng_t>
+void sequential_read_impl(benchmark::State & state, rng_t && rng)
+{
+    for (auto _ : state)
+    {
+        for (auto && outer : rng)
+        {
+            for (aa27 inner : outer)
+            {
+                benchmark::DoNotOptimize(inner.to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void sequential_read(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(100, 0, 0);
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<std::vector<aa27>> cpy = c | views::translate_join | views::to<std::vector<std::vector<aa27>>>;
+        sequential_read_impl(state, cpy);
+    }
+    else if constexpr (std::is_same_v<tag_t, translate_tag>)
+    {
+        auto v = c | views::translate | views::join;
+        sequential_read_impl(state, v);
+    }
+    else
+    {
+        auto v = c | views::translate_join;
+        sequential_read_impl(state, v);
+    }
+}
+
+BENCHMARK_TEMPLATE(sequential_read, baseline_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_join_tag);
+
+// ============================================================================
+//  random_access
+// ============================================================================
+
+template <typename rng_t>
+void random_access_impl(benchmark::State & state, rng_t && rng,
+                        std::vector<size_t> const & access_positions_outer,
+                        std::vector<std::vector<size_t> > const & access_positions_inner)
+{
+    for (auto _ : state)
+    {
+        for (size_t i = 0; i < access_positions_outer.size(); ++i)
+        {
+            for (size_t j = 0; j < access_positions_inner[i].size(); ++j)
+            {
+                benchmark::DoNotOptimize(rng[i][j].to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void random_access(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(200, 0, 0);
+
+    std::vector<size_t> access_positions_outer;
+    access_positions_outer.resize(200);
+    std::mt19937 gen(42);
+    std::uniform_int_distribution<size_t> dis_position_outer(0, 1000 - 1);
+
+    for (size_t i = 0; i < 200; ++i)
+        access_positions_outer[i] = dis_position_outer(gen);
+
+    std::vector<std::vector<size_t> > access_positions_inner;
+    access_positions_inner.resize(200);
+    for (size_t i = 0; i < 200; ++i)
+        access_positions_inner[i].resize(20);
+    std::uniform_int_distribution<size_t> dis_position_inner(0, 50 - 1);
```

Comment:
Why only between 0 and 49 if the sequences are 200 long?

Reply:
Because if the access is on the inner range it is on aminoacid level and the 200 nt DNA sequences get reduced to 1/3 of their length. Or am I missing something?

Reply:
Ah, of course! :clap: :+1:

Maybe add a comment?

Reply:
Sure :slightly_smiling_face: